pkg/postgres/postgres.go

```diff
@@ -180,3 +181,10 @@ func IsCheckViolation(err error) bool {
 	}
 	return false
 }
+
+func IsRaiseException(err error) bool {
+	if e, ok := err.(pgx.PgError); ok && e.Code == RaiseException {
```

Comment:
I wonder if this function should be generalized to take a Postgres error code constant instead? (could replace `IsCheckViolation` too)